File: 2D_amber/Assets/Scripts/JS/World/World.js
#pragma strict
internal var sound : SoundManager ;
internal var ui : MainUI ;
internal var dialog : DialogUI ;

function Awake(){
  sound = FindObjectOfType.<SoundManager>() ;
  ui = FindObjectOfType.<MainUI>() ;
  dialog = FindObjectOfType.<DialogUI>() ;
}

function StartTalk(storyID : int){
  Game.pause = true ;
  ui.gameObject.SetActive(false) ;
  dialog.enabled = true ;
  dialog.StartTalk(storyID) ;
}

function EndTalk(){
  Game.pause = false ;
  ui.gameObject.SetActive(true) ;
}

File: 2D_amber/Assets/Scripts/CSharp/World/World.cs
using UnityEngine;
using System.Collections;

public class World : MonoBehaviour {

	internal SoundManager sound ;
	internal MainUI ui ;
	internal DialogUI dialog ;

	void Awake(){
		sound = FindObjectOfType<SoundManager>() ;
		ui = FindObjectOfType<MainUI>() ;
		dialog = FindObjectOfType<DialogUI>() ;
	}
	
	public void StartTalk(int storyID){
		Game.pause = true ;
		ui.gameObject.SetActive(false) ;
		dialog.enabled = true ;
		dialog.StartTalk(storyID) ;
	}

	public void EndTalk(){
		Game.pause = false ;
		ui.gameObject.SetActive(true) ;
	}
}
